h/static/scripts/controllers/share-widget-controller.js

```diff
@@ -0,0 +1,212 @@
+'use strict';
+
+const Controller = require('../base/controller');
+
+const CONFIG_ATTR = 'share-widget-config';
+const TRIGGER_SELECTOR = `[${CONFIG_ATTR}]`;
+const WIDGET_SELECTOR = '.js-share-widget-owner';
+const TARGET_HREF_ATTR = 'share-target-href';
+const TARGET_HREF_SELECTOR = `[${TARGET_HREF_ATTR}]`;
+const CLIPBOARD_INPUT_SELECTOR = '.js-share-widget-clipboard';
+const PRIVATE_MSG_SELECTOR = '.js-share-widget-msg-private';
+const GROUP_MSG_SELECTOR = '.js-share-widget-msg-group';
+
+const ARROW_PADDING_RIGHT =  16;
+const ARROW_PADDING_BOTTOM =  5;
+
+let shareWidgetAttached = false;
+
+// Travel up the nodes parents to find a matching
+// element with the share-widget configuration values
+const _getParentWithConfig = (node) => {
+  let configNode = null;
+  while ((node = node.parentNode)) {
+    if (node === document.body) {
+      break;
+    }
+    if (node.hasAttribute(CONFIG_ATTR)) {
+      configNode = node;
+      break;
+    }
+  }
+  return configNode;
+};
+
+const _getOffset = (el) => {
```

Comment:
The `_` prefix on this helper function isn't needed. We use the `_` prefix on private class methods/variables because there isn't a way to avoid exporting them.